src/VisualStudio/Core/Def/Implementation/ProjectSystem/FileChangeWatcherProvider.cs

```diff
@@ -29,5 +29,18 @@ public FileChangeWatcherProvider(IThreadingContext threadingContext, [Import(typ
         }
 
         public FileChangeWatcher Watcher { get; }
+
+        // HACK HACK: this is to work around the SwitchToMainThread in the constructor above not
+        // being practical to run in unit tests. That SwitchToMainThread is working around a now-fixed
+        // bug in the shell where GetServiceAsync() might deadlock in the VS service manager
+        // if the UI thread was also dealing with the service at the same time. I'd remove the
+        // SwitchToMainThreadAsync right now instead of this doing this hack, but we're targeting this
+        // fix for a preview release that's too risky to do it in. Other options involve more extensive
```

Comment:
Do we want to file a tracking bug to do the right fix in 16.1?

Reply:
Ah right I didn't link to it in the comment but it's https://github.com/dotnet/roslyn/issues/33506